crates/store/re_video/src/demux/mod.rs

```diff
@@ -104,9 +109,23 @@ impl SamplesStatistics {
             .iter()
             .all(|s| s.decode_timestamp == s.presentation_timestamp);
 
+        let mut biggest_pts_so_far = Time::MIN;
+        let has_sample_highest_pts_so_far = samples
```

Comment:
I'm also a bit worried about how many times we're iterating over the samples here. I don't think LLVM will optimize this to one loop.

Reply:
tbh I'm not sure if it's better to combine everything to a single loop or to loop several times: individual loops might vectorize better

pure speculation until someone actually measures. Maybe I should. After all we might run this with _a lot_ of samples!

Reply:
It optimizes a lot better than I thought: https://godbolt.org/z/n3Yvzff9e

Reply:
neither is in the mood for simd though. probably because everything is 64bit integer

Reply:
leaving it be as is for now